services/vod-to-live/service/src/messaging/handlers/check-channel-job-status-handler.ts

```diff
@@ -0,0 +1,125 @@
+import { Broker, MessageInfo } from '@axinom/mosaic-message-bus';
+import { Logger, sleep } from '@axinom/mosaic-service-common';
+import {
+  CheckChannelJobStatusCommand,
+  CheckChannelJobStatusFailedEvent,
+  CheckChannelJobStatusSucceededEvent,
+  VodToLiveServiceMessagingSettings,
+} from 'media-messages';
+import { VirtualChannelApi } from 'src/domains';
+import urljoin from 'url-join';
+import { Config } from '../../common';
+import { AuthenticatedMessageHandler } from './authenticated-message-handler';
+
+export class CheckChannelJobStatusHandler extends AuthenticatedMessageHandler<CheckChannelJobStatusCommand> {
+  private logger: Logger;
+  constructor(
+    config: Config,
+    private virtualChannelApi: VirtualChannelApi,
+    private broker: Broker,
+  ) {
+    super(
+      VodToLiveServiceMessagingSettings.CheckChannelJobStatus.messageType,
+      config,
+    );
+    this.logger = new Logger({
+      config: this.config,
+      context: CheckChannelJobStatusHandler.name,
+    });
+  }
+  async onMessage(
+    { channel_id, seconds_elapsed_while_waiting }: CheckChannelJobStatusCommand,
+    message: MessageInfo,
+  ): Promise<void> {
+    // request status
+    const channelStatus = await this.virtualChannelApi.getChannelStatus(
+      channel_id,
+    );
+
+    if (channelStatus.status === 'Success') {
+      const dashUrl = urljoin(
+        this.config.virtualChannelOriginBaseUrl,
+        `${channel_id}.isml`,
+        '.mpd',
+      );
+      const hlsUrl = urljoin(
+        this.config.virtualChannelOriginBaseUrl,
+        `${channel_id}.isml`,
+        '.m3u8',
+      );
+      this.logger.log({
+        message: `Channel ${channel_id} successfully went live.`,
+        details: {
+          dashUrl,
+          hlsUrl,
+        },
+      });
+      await this.broker.publish<CheckChannelJobStatusSucceededEvent>(
+        VodToLiveServiceMessagingSettings.CheckChannelJobStatusSucceeded
+          .messageType,
+        {
+          channel_id,
+          dash_stream_url: dashUrl,
+          hls_stream_url: hlsUrl,
+        },
+        { auth_token: message.envelope.auth_token },
+      );
+      return;
+    } else if (channelStatus.status === 'Failed') {
+      const uspErrors = channelStatus.details
+        .filter((d) => d.status === 'Failed')
+        .map(
+          (failed) => `Failed task: ${failed.name}. Details: ${failed.details}`,
+        );
+
+      this.logger.error({
+        message: `Channel ${channel_id} failed to go live.`,
+        details: {
+          uspErrors,
+        },
+      });
+
+      await this.broker.publish<CheckChannelJobStatusFailedEvent>(
```

Comment:
Should some service receive this (CheckChannelJobStatusFailedEvent) message and act accordingly?